js/tests/friend-support_test.mjs

```diff
@@ -0,0 +1,125 @@
+import { once } from 'node:events'
+import * as cp from 'node:child_process'
+import { mkdir, writeFile, chmod } from 'fs/promises'
+import { tmpdir } from 'node:os'
+import { join } from 'path'
+
+export const tests = []
+const fetch = _fetch // to redefine the real fetch
+
+const port = 5000
+
+export const setup = async ({ randStr }) => {
+  const dir = '.'
+
+  await mkdir(`${dir}/guests`, { recursive: true })
+
+  const randLastName = randStr()
+
+  const createFilesIn = ({ files, dirPath }) =>
+    Promise.all(
+      files.map(([fileName, content]) =>
+        writeFile(`${dirPath}/${fileName}`, JSON.stringify(content)),
+      ),
+    )
+
+  const sendRequest = async (path, options) => {
+    const response = await fetch(`http://localhost:${port}${path}`, options)
+    const { status, statusText, ok } = response
+    const headers = Object.fromEntries(response.headers)
+    let body = ''
+    try {
+      body = await response.json()
+    } catch (err) {
+      body = err
+    }
+    return { status, body, headers, randLastName }
+  }
+
+  return { tmpPath: dir, createFilesIn, sendRequest }
+}
+
+tests.push(async ({ path, ctx }) => {
+  ctx.server = cp.spawn('node', [`${path}`])
+  const message = await Promise.race([
+    once(ctx.server.stdout, 'data'),
+    Promise.race([
+      once(ctx.server.stderr, 'data').then(String).then(Error),
+      once(ctx.server, 'error'),
+    ]).then(x => Promise.reject(x)),
+  ])
+  return message[0].toString().includes(port)
+})
+
+tests.push(async ({ eq, ctx, randStr }) => {
```

Comment:
is randStr being used?

Reply:
I will actually use it to randomize the message to avoid hard-coded solutions. Thank you for the catch :)